### PR TITLE
lightning: rename and move some struct

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #40499

Problem Summary:

### What is changed and how it works?
- rename some struct to better reflect what it does, such as  tableRestore  to tableImporter
- move chunk related code into separate file

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/importer/chunk_process.go [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+	"github.com/pingcap/tidb/keyspace"
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/types"
+	"go.uber.org/zap"
+)
+
+// chunkProcessor process data chunk
+// for local backend it encodes and writes KV to local disk
+// for tidb backend it transforms data into sql and executes them.
+type chunkProcessor struct {
```
Comment (D3Hunter) [inline br/pkg/lightning/importer/chunk_process.go:46]: moved from previous restore.go

File: br/pkg/lightning/importer/chunk_process_test.go [hunk 2]
```diff
@@ -780,3 +780,32 @@ func TestCompressChunkRestore(t *testing.T) {
 	err = cr.parser.ReadRow()
 	require.Equal(t, io.EOF, errors.Cause(err))
 }
+
+func TestGetColumnsNames(t *testing.T) {
```
Comment (D3Hunter) [inline br/pkg/lightning/importer/chunk_process_test.go:784]: moved from previous restore_table_test.go

File: br/pkg/lightning/importer/table_import.go [hunk 3]
```diff
@@ -89,12 +91,121 @@ func NewTableRestore(
 	}, nil
 }
 
-func (tr *TableRestore) Close() {
+func (tr *TableImporter) importTable(
```
Comment (D3Hunter) [inline br/pkg/lightning/importer/table_import.go:94]: moved from previous restore.go

**Review by lance6716 — APPROVED** [review]

File: br/pkg/lightning/importer/BUILD.bazel [hunk 4]
```diff
@@ -91,23 +92,21 @@ go_library(
 )
 
 go_test(
-    name = "restore_test",
-    timeout = "short",
```
Comment (hawkingrei) [inline br/pkg/lightning/importer/BUILD.bazel:95]: ```timeout``` and ```flaky``` isn't removed.

**Review by hawkingrei — APPROVED** [review]

**Comment by D3Hunter** [comment]
lightning ci pass
https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/15404/pipeline/114/

**Pull request merged**